plugins/woocommerce/src/Internal/Fulfillments/OrderFulfillmentsRestController.php

```diff
@@ -0,0 +1,1084 @@
+				'description' => __( 'The meta data array.', 'woocommerce' ),
+				'type'        => 'array',
+				'required'    => true,
+				'items'       => array(
+					'description' => __( 'The meta data object.', 'woocommerce' ),
+					'type'        => 'object',
+					'schema'      => $this->get_schema_for_meta_data(),
+				),
+			),
+		);
+
+		return $schema;
+	}
+
+	/**
+	 * Get the arguments for the get tracking number details endpoint.
+	 *
+	 * @return array
+	 */
+	private function get_args_for_get_tracking_number_details(): array {
+		return array(
+			'tracking_number' => array(
+				'description' => __( 'The tracking number to look up.', 'woocommerce' ),
+				'type'        => 'string',
+				'required'    => true,
+				'context'     => array( 'view', 'edit' ),
+			),
+		);
+	}
+
+	/**
+	 * Get the schema for the get tracking number details endpoint.
+	 *
+	 * @return array
+	 */
+	private function get_schema_for_get_tracking_number_details(): array {
+		$schema               = $this->get_base_schema();
+		$schema['title']      = __( 'The tracking number details response.', 'woocommerce' );
+		$schema['properties'] = array(
+			'tracking_number_details' => array(
+				'description' => __( 'The tracking number details.', 'woocommerce' ),
+				'type'        => 'object',
+				'required'    => true,
+				'properties'  => array(
+					'tracking_number'   => array(
+						'description' => __( 'The tracking number.', 'woocommerce' ),
+						'type'        => 'string',
+						'required'    => true,
+					),
+					'shipping_provider' => array(
+						'description' => __( 'The shipping provider.', 'woocommerce' ),
+						'type'        => 'string',
+						'required'    => true,
+					),
+					'tracking_url'      => array(
+						'description' => __( 'The tracking URL.', 'woocommerce' ),
+						'type'        => 'string',
+						'required'    => true,
+					),
+				),
+			),
+		);
+		return $schema;
+	}
+
+	/**
+	 * Get the base schema for the fulfillment with a read context.
+	 *
+	 * @return array
+	 */
+	private function get_read_schema_for_fulfillment() {
+		return array(
+			'fulfillment_id' => array(
+				'description' => __( 'Unique identifier for the fulfillment.', 'woocommerce' ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+			'entity_type'    => array(
+				'description' => __( 'The type of entity for which the fulfillment is created.', 'woocommerce' ),
+				'type'        => 'string',
+				'required'    => true,
+				'context'     => array( 'view', 'edit' ),
+			),
+			'entity_id'      => array(
+				'description' => __( 'Unique identifier for the entity.', 'woocommerce' ),
+				'type'        => 'string',
+				'required'    => true,
+				'context'     => array( 'view', 'edit' ),
+			),
+			'status'         => array(
+				'description' => __( 'The status of the fulfillment.', 'woocommerce' ),
+				'type'        => 'string',
+				'default'     => 'unfulfilled',
```

Comment:
Are the possible values just `fulfilled` and `unfulfilled`? I think we could use an `enum` property here.

Reply:
Possible values for that will be dynamic, as the user will be able to add more with a filter provided in a separate PR.

Reply:
Got it, then it's fine, thanks!